content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -846,25 +848,59 @@ private int getSimilarityReportStatus(String reportId) throws Exception {
 
 	private String getSubmissionId(ContentReviewItem item, String fileName, Site site, Assignment assignment) {
 		String userID = item.getUserId();
-
+		String submitterID = null;
+		List<User> submissionOwners = new ArrayList<User>();
+		try {
+			AssignmentSubmission currentSubmission = assignmentService.getSubmission(assignment.getId(), item.getUserId());
+			String submittedById = currentSubmission.getProperties().get(AssignmentConstants.SUBMITTER_USER_ID);
+			if(StringUtils.isNotEmpty(submittedById) && !submittedById.equals(userID)) {
+				submitterID = submittedById;
+			}
+			//find owner metadata:
+			Set<String> ownerIds = currentSubmission.getSubmitters().stream()
+				.map(s -> s.getSubmitter())
+				.collect(Collectors.toSet());
+			ownerIds.add(userID);
+			if(StringUtils.isNotEmpty(submitterID)) {
+				ownerIds.add(submitterID);
+			}
+			for(String ownerId : ownerIds) {
+				try {
+					submissionOwners.add(userDirectoryService.getUser(ownerId));
+				}catch(Exception e) {
+					log.error(e.getMessage(), e);
+				}
+			}
+		}catch(Exception e) {
+			log.error(e.getMessage(), e);
+		}
 		String submissionId = null;
 		try {
 
 			// Build header maps
 			Map<String, Object> data = new HashMap<String, Object>();
 			data.put("owner", userID);
+			if(StringUtils.isNotEmpty(submitterID)) {
+				data.put("submitter", submitterID);	
+			}
 			data.put("title", fileName);
-			Instant eulaTimestamp = getUserEULATimestamp(userID);
-			String eulaVersion = getUserEULAVersion(userID);
-			if(eulaTimestamp != null && StringUtils.isNotEmpty(eulaVersion)) {
-				Map<String, Object> eula = new HashMap<String, Object>();
-				eula.put("accepted_timestamp", eulaTimestamp.toString());
-				eula.put("language", getUserEulaLocale(userID));
-				eula.put("version", eulaVersion);
-				data.put("eula", eula);
+			String eulaUserId = StringUtils.isNotEmpty(submitterID) ? submitterID : userID;
+			Instant eulaTimestamp = getUserEULATimestamp(eulaUserId);
+			String eulaVersion = getUserEULAVersion(eulaUserId);
+			if(eulaTimestamp == null || StringUtils.isEmpty(eulaVersion)) {
+				//best effort to make sure the user has a EULA acceptance timestamp, but if not
+				//add a warning in the logs and continue so that the report will still generate
+				eulaTimestamp = Instant.now();
+				eulaVersion = getEndUserLicenseAgreementVersion();
+				log.warn("EULA not found for user: " + eulaUserId + ", contentId: " + item.getId());
 			}
-			if(assignment != null) {
-				Map<String, Object> metadata = new HashMap<String, Object>();
+			Map<String, Object> eula = new HashMap<String, Object>();
+			eula.put("accepted_timestamp", eulaTimestamp.toString());
+			eula.put("language", getUserEulaLocale(eulaUserId));
+			eula.put("version", eulaVersion);
+			data.put("eula", eula);
```

Comment:
It appears that the nested Map<String, Object> were flattened it could just be a Map<String, String>
If so maybe partitioning the maps using the key so that your querying the keys to return subsets of the data contained within, for example to query all eula keys you could do:
`Map<String, String> eulaData = data.entrySet().stream().filter(e -> e.getKey().startsWith("eula_")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));`

I only provide this as another way to think about storing this kind of data in a single map vs the nesting :)